src/workers/airtable-sync/actions/updateMessageContent.test.js

```diff
@@ -0,0 +1,99 @@
+jest.mock("~slack/channels", () => ({ getExistingMessage: jest.fn() }));
+jest.mock("~slack/webApi", () => ({ chat: { update: jest.fn() } }));
+
+require("~strings/i18nextInit");
+const { Record } = require("airtable");
+const {
+  chat: { update: updateFn }
+} = require("~slack/webApi");
+const { getExistingMessage: getExistingMessageFn } = require("~slack/channels");
+const updateMessageContent = require("./updateMessageContent");
+const requests = require("~airtable/tables/requests");
+
+describe('When request has been posted with status "Delivery Needed"', () => {
+  let requestRecord;
+
+  describe("and the record does not have cross streets", () => {
+    beforeEach(() => {
+      getExistingMessageFn.mockResolvedValue({
+        text: `*Household Size*: n/a
+*Cross Streets*: <https://crownheightsma.herokuapp.com/delivery-needed?request=RN7SQ3|>
+*Description*: No notes`
+      });
+
+      requestRecord = new Record(requests.tableName, "id-1", {
+        [requests.fields.status]: requests.fields.status_options.deliveryNeeded,
+        [requests.fields.crossStreetFirst]: null,
+        [requests.fields.crossStreetSecond]: null
+      });
+
+      // this function is added by airtable-change-detector
+      requestRecord.getMeta = () => ({
+        slack_ts: "some-slack-ts",
+        slack_channel: "some-slack-channel"
+      });
+    });
+
+    describe('and the status is changed to "Delivery Assigned"', () => {
+      beforeEach(() => {
+        updateFn.mockClear();
+
+        requestRecord.set(
+          requests.fields.status,
+          requests.fields.status_options.deliveryAssigned
+        );
+      });
+
+      test("the cross streets line is removed from the post", async () => {
+        await updateMessageContent(requestRecord);
+
+        expect(updateFn.mock.calls[0][0].text).not.toMatch(/\*Cross Streets\*/);
```

Comment:
non-blocking for merge, but might be more readable to use `expect(updateFn).toHaveBeenCalledWith(expect.stringMatching(<regex>))`

i have not tried doing `expect.not.stringMatching(<regex>)` but maybe worth trying.

Reply:
Since the string is in a nested object, I think the alternative would be more like `expect(updateFn).toHaveBeenCalledWith(expect.objectMatching({text: expect.not.stringMatching(<regex>)}))` which I don't necessarily find to be more readable than it is now